File: Algorithm/Leetcode/0000_0100/0011_0020/0014.py
# 1528. Shuffle String

"""https://leetcode.com/problems/shuffle-string/discuss/803446/Python-3-one-liner"""

class Solution:
    def restoreString(self, s: str, indices: List[int]) -> str:
        return ''.join([v for (_,v) in sorted(zip(indices, s))])

File: Algorithm/Leetcode/0000_0100/0011_0020/0014.cpp
// 1528. Shuffle String


class Solution {
public:
    string restoreString(string s, vector<int>& indices) {
        int n = s.size();
        string output;
        output.resize(n);
        for(int i = 0; i< n; ++i)
        {
            output[indices[i]] = s[i];
        }
        return output;
    }
};